Source/WebCore/rendering/RenderGrid.cpp

```diff
@@ -1022,6 +1022,16 @@ LayoutUnit RenderGrid::masonryContentSize() const
     return m_masonryLayout.gridContentSize();
 }
 
+Vector<LayoutRect> RenderGrid::gridItemsLayoutRects()
+{
+    Vector<LayoutRect> items;
```

Comment:
Should we assert here to make sure the caller does not call this function on a grid that needs to go through layout? Otherwise they would be getting stale geometry for the renderers

Reply:
Discussed with Sammy about the assert.

This is somewhat more of a discussion who owns the responsibility of ensuring that the data is not stale.

Will handle this in a future patch.